srtcore/core.cpp

```diff
@@ -2290,6 +2292,38 @@ static inline bool NextExtensionBlock(uint32_t*& w_begin, uint32_t* next, size_t
     return true;
 }
 
+void SrtExtractHandshakeExtensions(const char* bufbegin, size_t buflength,
```

Comment:
This function isn't used in the SRT code itself, but placing it here is necessary for applications that would like to make use of a direct protocol processing. It's needed here because it reaches out to parts of the code that are otherwise inaccessible for the external applications.